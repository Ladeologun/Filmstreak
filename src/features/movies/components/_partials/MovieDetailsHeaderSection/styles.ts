import { StyleSheet, Platform, StatusBar, Dimensions } from "react-native";
import { COLORS } from "~styles";


const { height } = Dimensions.get('window')
const styles = StyleSheet.create({
    
    background:{
        width:"100%",
        height:height < 700 ? height* 0.5:height*0.65
    },
    
    title:{
        color:COLORS.WHITE,
        marginTop:8,
        // fontFamily: "Avenir", 
        fontSize: 30, 
        lineHeight: 36
    },
    gradientOvelay: {
        position: 'absolute',
    bottom: 0,
    width: '100%',
    height: 200,
    backgroundColor: 'rgba(15, 17, 29, 0.7)', // Dark semi-transparent color
    zIndex: 1,
    }

})

export default styles;