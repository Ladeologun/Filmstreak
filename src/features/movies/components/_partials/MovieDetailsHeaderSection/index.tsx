import { Image, ImageBackground, Text, View } from "react-native";
import styles from "./styles";
import DetailsHeader from "../DetailsHeader";
import MovieDetailsCTA from "../MovieDetailsCTA/Index";

interface Iprops {
    poster: string;
    title: string;
}


const MovieDetailsHeaderSection:React.FC<Iprops> = ({poster, title}) => {

    return(
        <ImageBackground source={{uri:poster}}
            style={styles.background}
            resizeMode="cover"
        >
            <View style={{flex:1}}>
                <DetailsHeader />

                <View style={{flex:1,justifyContent:"flex-end"}}>
                    
                    <View
                        style={
                            {width:"100%",height:200,alignItems:"center",justifyContent:"flex-end", backgroundColor:"#0F111D"}
                        }
                    >
                        <MovieDetailsCTA />
                        <Text style={styles.title}>{title}</Text>
                    </View>
                </View>
            </View>
        </ImageBackground>
    )
}

export default MovieDetailsHeaderSection;