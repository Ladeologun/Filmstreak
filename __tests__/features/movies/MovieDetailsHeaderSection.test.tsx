import renderer from 'react-test-renderer';
import {test,expect} from '@jest/globals';
import MovieDetailsHeaderSection from '~movies/components/_partials/MovieDetailsHeaderSection';


test('renders correctly', () => {
    const tree = renderer.create(
        <MovieDetailsHeaderSection
            poster={"https://example.com/backposter.jpg"}
            showWatchlistWalkthrough={false}
            title='now playing'
            handleWalkthroughClose={()=>{}}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});
