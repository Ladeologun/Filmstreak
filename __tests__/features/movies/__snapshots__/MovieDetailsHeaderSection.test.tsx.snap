// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "height": 867.1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "uri": "https://example.com/backposter.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": 867.1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 40,
            "paddingHorizontal": 24,
          },
          {
            "marginTop": 4,
            "paddingHorizontal": 10,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.4)",
            "borderBottomLeftRadius": 20,
            "borderBottomRightRadius": 20,
            "borderRadius": 20,
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/drawables/left-arrow.png",
            }
          }
          style={
            {
              "height": 20,
              "tintColor": "#ffffff",
              "width": 20,
            }
          }
        />
      </View>
      <View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0.4)",
                "borderRadius": 20,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": 50,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/drawables/wish-list.png",
                }
              }
              style={
                {
                  "height": 25,
                  "width": 25,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(15, 17, 29, 0.2)",
            "height": 200,
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontFamily": "PermanentMarker-Regular",
              "fontSize": 30,
              "lineHeight": 36,
              "marginTop": 8,
            }
          }
        >
          now playing
        </Text>
      </View>
    </View>
  </View>
</View>
`;
